docs: added instructions for installing a newer rkt in coreos

dgonyeo opened the pull request:
Added Documentation/install-rkt-in-coreos.md, which details a systemd
oneshot unit file that downloads and extracts rkt to /opt. Added a
reference to this file in Documentation/distributions.md.

Fixes https://github.com/coreos/rkt/issues/2601#issuecomment-218463662.

euank commented on Documentation/install-rkt-in-coreos.md:
```diff
@@ -0,0 +1,31 @@
+ExecStart=/usr/bin/curl -L -o /opt/rkt.tar.gz <rkt-url>
+ExecStart=/usr/bin/curl -L -o /opt/stage1-coreos-linux-amd64.aci <rkt-stage1-url>
+ExecStart=/usr/bin/curl -L -o /opt/stage1-coreos-linux-amd64.aci.asc <rkt-stage1-asc-url>
+ExecStart=/usr/bin/tar xf /opt/rkt.tar.gz -C /opt
```
Note, this leaves rkt in `/opt/rkt-vX.Y.Z/rkt`, and the tarball already included the stage1 in any case. It's less useful for rkt to be in a directory that varies by version because you now have to change everything that reverences the rkt binary path (e.g. your PATH).

A possible solution to this is `tar --strip-components=1 -xf rkt.tar.gz -C /opt/rkt/` which leaves you with a consistent `/opt/rkt/rkt`.

iaguis replied:
> and the tarball already included the stage1 in any case

Yep, and I believe that the stage1 included in the rkt tar file is the one that will be used. Note that we're not checking the signatures of the rkt tar... 

> A possible solution to this is tar --strip-components=1 -xf rkt.tar.gz -C /opt/rkt/ which leaves you with a consistent /opt/rkt/rkt.

+1

jonboulle replied:
> Note that we're not checking the signatures of the rkt tar...

please add this...

chancez commented:
Can we perhaps just retrieve rkt using rkt? The newer one could just get it's own CAS. Something like:

```
mkdir -p /var/lib/rkt-1.7.0
rkt fetch coreos.com/rkt:1.7.0
rkt run --volume var-rkt,kind=host,source=/var/lib/rkt-1.7.0 --mount volume=data,target=/var/lib/rkt coreos.com/rkt:1.7.0
```

Could optionally mount in the host's /etc/rkt to share config.

jonboulle commented:
Bump @dgonyeo

dgonyeo commented:
I just:
- changed the tar command as per @euank's suggestion to place the binary at `/opt/rkt/rkt`
- Added the curl and gpg commands to fetch CoreOS's app signing key, and verify the fetched rkt tar
- removed the curl command to fetch a stage1, since they're included in the fetched tarball

euank commented on Documentation/install-rkt-in-coreos.md:
```diff
@@ -0,0 +1,38 @@
+Type=oneshot
+RemainAfterExit=yes
+ExecStart=/usr/bin/mkdir -p /opt/rkt
+ExecStart=/usr/bin/curl -L -o /opt/rkt.tar.gz <rkt-url>
```
`--silent`? Is there a reason not to?

dgonyeo replied:
Nope. I'll go add that.

iaguis commented:
The users would have to run rkt with the full path (`/opt/rkt/rkt`) but I guess that's fine. LGTM

iaguis commented:
Can you rebase to master?

dgonyeo commented:
Done

jonboulle commented on Documentation/install-rkt-in-coreos.md:
```diff
@@ -0,0 +1,38 @@
+oneshot systemd unit can be used to download and install an alternate version
+on boot.
+
+The following unit will use curl to download rkt, it's signature, and the
```
its

iaguis commented:
LGTM after jon's nit

dgonyeo commented:
s/it's/its/ is done
